### PR TITLE
Add `CytoDataFrame` dependency; remove migrated code

### LINKED ISSUE #108: Migrate `CytoDataFrame` to separate project
We plan to migrate `CytoDataFrame` to a separate project to help keep things loosely coupled where possible and give primary focus to this project in the QC domain.

### PR DESCRIPTION
**Pull request by d33bs**
# Description

This PR adds `CytoDataFrame` as a dependency to the project, which is now a migrated independent project. Alongside this I removed code which has been migrated and added a check for dead code through `vulture` and pre-commit. Some minor formatting changes were also applied through an update to pre-commit checks.

Closes #108 

## What kind of change(s) are included?

- [ ] Documentation (changes docs or other related content)
- [ ] Bug fix (fixes an issue).
- [x] Enhancement (adds functionality).
- [ ] Breaking change (these changes would cause existing functionality to not work as expected).

# Checklist

Please ensure that all boxes are checked before indicating that this pull request is ready for review.

- [x] I have read and followed the [CONTRIBUTING.md](CONTRIBUTING.md) guidelines.
- [x] I have searched for existing content to ensure this is not a duplicate.
- [x] I have performed a self-review of these additions (including spelling, grammar, and related).
- [x] These changes pass all pre-commit checks.
- [ ] I have added comments to my code to help provide understanding
- [ ] I have added a test which covers the code changes found within this PR
- [x] I have deleted all non-relevant text in this pull request template.

### CODE REVIEW

File: media/coverage-badge.svg
Comment (jenna-tomkinson) [inline media/coverage-badge.svg:1]: Ouch! What happened!

File: pyproject.toml [hunk 1]
```diff
@@ -29,24 +29,8 @@ pyyaml = "^6.0.1"
...
-  { version = "^0.19.3", python = "<3.9" },
-  { version = ">0.19.3", python = ">=3.9" },
-]
-# dependency of scikit-image
-pywavelets = [
-  { version = "^1.4.1", python = "<3.9" },
-  { version = ">1.4.1", python = ">=3.9" },
-]
-# used for image modifications in cytodataframe
-opencv-python = "^4.10.0.84"
```
Comment (jenna-tomkinson) [inline pyproject.toml:49]: Are these removed because they only applied to CytoDataFrame?

**Review by jenna-tomkinson — APPROVED** [review]
LGTM! 🎉

File: media/coverage-badge.svg
Comment (d33bs) [reply media/coverage-badge.svg:1]: As we removed code and modules I believe the denominator for the calculations caused a drop here. We'll address it as we go and this isn't a bad number!

File: pyproject.toml (hunk 1 above)
Comment (d33bs) [reply pyproject.toml:49]: Thanks for asking - yes, these were dropped as I found they were only required by CytoDataFrame.

**Comment by d33bs** [comment]
Thanks @jenna-tomkinson ! Merging this in.

**Pull request merged**